tests/Unit/API/Google/MerchantTest.php

```diff
@@ -0,0 +1,321 @@
+		];
+
+		$list_response->expects( $this->any() )
+			->method( 'getResources' )
+			->willReturn( $product_list );
+
+		$list_response->expects( $this->any() )
+			->method( 'getNextPageToken' )
+			->will(
+				$this->onConsecutiveCalls(
+					'token',
+					'token',
+					null
+				)
+			);
+
+		$this->service->products->expects( $this->any() )
+			->method( 'listProducts' )
+			->willReturn( $list_response );
+
+		$products = $this->merchant->get_products();
+		$this->assertCount( count( $product_list ) * 2, $products );
+	}
+
+	public function test_claim_website() {
+		$this->assertTrue( $this->merchant->claimwebsite() );
+	}
+
+	public function test_claimwebsite_error() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'claimwebsite' )
+			->will(
+				$this->throwException(
+					new GoogleException()
+				)
+			);
+
+		$this->expectException( Exception::class );
+		$this->merchant->claimwebsite();
+	}
+
+	public function test_website_already_claimed() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'claimwebsite' )
+			->will(
+				$this->throwException(
+					new GoogleException( 'claimed', 403 )
+				)
+			);
+
+		$this->expectException( Exception::class );
+        $this->expectExceptionCode( 403 );
+		$this->merchant->claimwebsite();
+	}
+
+	public function test_get_account() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'get' )
+			->willReturn( $this->createMock( Account::class ) );
+
+		$this->assertInstanceOf( Account::class, $this->merchant->get_account() );
+	}
+
+	public function test_get_account_failure() {
+		$this->service->accounts->expects( $this->any() )
+			->method( 'get' )
+			->will(
+				$this->throwException(
+					new GoogleException( 'error', 400 )
+				)
+			);
+
+		$this->expectException( MerchantApiException::class );
+        $this->expectExceptionCode( 400 );
+		$this->merchant->get_account();
+	}
+
+	public function test_get_accountstatuses() {
+		$this->service->accountstatuses = $this->createMock( Accountstatuses::class );
+		$this->service->accountstatuses->expects( $this->any() )
+			->method( 'get' )
+			->willReturn( $this->createMock( AccountStatus::class ) );
+
+		$this->assertInstanceOf( AccountStatus::class, $this->merchant->get_accountstatus() );
+	}
+
+	public function test_get_accountstatus_failure() {
+		$this->service->accountstatuses = $this->createMock( Accountstatuses::class );
+		$this->service->accountstatuses->expects( $this->any() )
+			->method( 'get' )
+			->will(
+				$this->throwException(
+					new GoogleException( 'error', 400 )
+				)
+			);
+
+		$this->expectException( Exception::class );
+        $this->expectExceptionCode( 400 );
+		$this->merchant->get_accountstatus();
+	}
+
+	public function test_get_productstatuses_batch() {
+		$this->service->productstatuses = $this->createMock( Productstatuses::class );
+
+		$this->service->productstatuses->expects( $this->any() )
+			->method( 'custombatch' )
```

Comment:
We could add an assertion here to cover the code that generates the `$entries` array e.g:

```suggestion
			->method( 'custombatch' )
			->with( $this->callback( function( $request ) {
				/** @var ProductstatusesCustomBatchRequest $request */
				$this->assertEquals(
					[
						'batchId'    => 3,
						'productId'  => 3,
						'method'     => 'GET',
						'merchantId' => 0,
					],
					$request->getEntries()[2]
				);

				return true;
			} ) )
```